samcli/local/apigw/local_apigw_service.py

```diff
@@ -413,6 +414,31 @@ def _get_current_route(self, flask_request):
 
         return route
 
+    @staticmethod
+    def _response_cors_headers(flask_request, cors_headers):
+        if "Access-Control-Allow-Origin" not in cors_headers:
+            return cors_headers
+
+        cors_origins = cors_headers["Access-Control-Allow-Origin"]
+        # unset this header due to restrictive manner
```

Comment:
Why unset this? It looks like it should be removed in the case that `response_allowed_origin` is not set. If so, I would move this to be closer to line 437. I think it just make it a little easier to read since the mutation of the `cors_headers` is all in one spot.

Reply:
Hi @jfuss , thank you for taking your time to review this

we unset and set again if a match is found (restrictive manner)

if multiple domains are allowed, we only send back 1 allowed domain in our response header
if all domains are allowed, we also only send back  1 allowed domain in our response header

we do not return this header (implying a deny) if no matches is found

Reply:
hi @jfuss please also refer to my below comment:
https://github.com/aws/aws-sam-cli/pull/4991#discussion_r1174612400